src/test/netbase_tests.cpp

```diff
@@ -395,6 +395,51 @@ BOOST_AUTO_TEST_CASE(netpermissions_test)
     BOOST_CHECK(NetWhitebindPermissions::TryParse("@1.2.3.4:32", whitebindPermissions, error));
     BOOST_CHECK_EQUAL(whitebindPermissions.m_flags, PF_NONE);
 
+    // "noban" implies "download"
+    BOOST_CHECK(NetWhitebindPermissions::TryParse("noban@1.2.3.4:32", whitebindPermissions, error));
+    BOOST_CHECK_EQUAL(whitebindPermissions.m_flags, PF_NOBAN);
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_DOWNLOAD);
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_NOBAN);
+    BOOST_CHECK(NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_DOWNLOAD));
+    BOOST_CHECK(NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_NOBAN));
+
+    NetPermissions::ClearFlag(whitebindPermissions.m_flags, PF_DOWNLOAD);
+    BOOST_CHECK(!(whitebindPermissions.m_flags & PF_DOWNLOAD));
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_NOBAN);
+    BOOST_CHECK(!NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_DOWNLOAD));
+    BOOST_CHECK(!NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_NOBAN));
+
+    // "noban,download" is equivalent to "noban"
+    BOOST_CHECK(NetWhitebindPermissions::TryParse("noban,download@1.2.3.4:32", whitebindPermissions, error));
+    BOOST_CHECK_EQUAL(whitebindPermissions.m_flags, PF_NOBAN);
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_DOWNLOAD);
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_NOBAN);
+    BOOST_CHECK(NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_DOWNLOAD));
+    BOOST_CHECK(NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_NOBAN));
+
+    // "download,noban" is equivalent to "noban"
+    BOOST_CHECK(NetWhitebindPermissions::TryParse("download,noban@1.2.3.4:32", whitebindPermissions, error));
+    BOOST_CHECK_EQUAL(whitebindPermissions.m_flags, PF_NOBAN);
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_DOWNLOAD);
+    BOOST_CHECK(whitebindPermissions.m_flags & PF_NOBAN);
+    BOOST_CHECK(NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_DOWNLOAD));
+    BOOST_CHECK(NetPermissions::HasFlag(whitebindPermissions.m_flags, PF_NOBAN));
```

Comment:
This is repeated 3 times: for `noban`, `noban,download` and `download,noban`. Could avoid the repetitions with something like:

```cpp
NetWhitebindPermissions noban;
NetWhitebindPermissions noban_download;
NetWhitebindPermissions download_noban;

TryParse("noban@...", noban);
...all 6 checks...

TryParse("noban,download@...", noban_download);
BOOST_CHECK_EQUAL(noban_download, noban); // just this instead of the 6 repeated checks

TryParse("download,noban@...", download_noban);
BOOST_CHECK_EQUAL(download_noban, noban); // just this instead of the 6 repeated checks
```

Reply:
done